feature: introduce config file for JPK tags

ns-rse opened the pull request:
Off the back of #118 (thanks @derollins) I've parameterised the JPK tags that were introduced into
`AFMReader/default_config.yaml` and load these by default when using `AFMReader.load_jpk()`. A parameter is introduced
to that function (`confi_path`) which allows users to specify their own configuration file should they wish to use alternative tags.

- Copies the `read_yaml()` function and tests from TopoStats.
- Brief description of configuration file and structure added to `docs/usage.md` (where some tables were also aligned by
  Emacs).

Exiting `tests/test_jpk.py` pass.

I envisage this being an extensible and flexible way of users being able to handle parameters changing the array of file types that 
exist now and in the future and it should be easy to extend to other file formats.

derollins commented on AFMReader/default_config.yaml:
Looks good, maybe to make it more complete another parameter for the slot size could be added. This is 48 and is used repeatedly to count through the 'slots' in the tiff metadata. 
suggestion:
```
jpk:
  n_slots: 32896
  default_slot: 32897
  first_slot_tag: 32912
  first_scaling_type: 32931
  first_scaling_name: 32932
  first_offset_name: 32933
  channel_name: 32848
  trace_retrace: 32849
  grid_ulength: 32834
  grid_vlength: 32835
  grid_ilength: 32838
  grid_jlength: 32839
  slot_size: 48
```

derollins commented on AFMReader/jpk.py:
```diff
@@ -79,9 +73,9 @@ def _get_z_scaling(tif: tifffile.tifffile, channel_idx: int) -> tuple[float, flo
-                ):
+                if tag_name_float >= (  # pylint: disable=chained-comparison
+                    int(jpk_tags["first_slot_tag"]) + (n_slots * 48)
+                ) and tag_name_float < (int(jpk_tags["first_slot_tag"]) + ((n_slots + 1) * 48)):
```
Here is the first use of `48` as the `slot_size`

derollins commented on docs/usage.md:
```diff
@@ -11,23 +11,45 @@ integrate it into your workflow.
+|             | `grid_ulength`       | `32834`       |
+|             | `grid_vlength`       | `32835`       |
+|             | `grid_ilength`       | `32838`       |
+|             | `grid_jlength`       | `32839`       |
```
Add `slot_size` here if we choose to.

derollins reviewed (commented):
Only suggestion is to add an extra value for completeness, have done it myself locally but don't know if I can change it here.

ns-rse commented:
Brilliant, thanks for the feedback @derollins :+1: 

I've added in `slot_size` to the configuration file, updated `AFMReader/jpk.py` to use this parameter and included it in the documentation.

derollins commented on docs/usage.md:
```diff
@@ -11,23 +11,46 @@ integrate it into your workflow.
+
+A default configuration is included in the package which contains options for loading different file formats. This is a
+[YAML](https://yaml.org) file with keys for each file type and the relevant key/value pair nested within. The table
+below details the current configuration options. If loading a file format supports a custom configuration file then it
```
How is `config_path` used to specify an alternative config file?

derollins reviewed (commented):
Captures all the .jpk file tags used to read the files and has a clear config file for the values of these tags.  I'm just unsure about how alternative config files can be used, since this is a library rather than an executable programme.

derollins commented on docs/usage.md:
```diff
@@ -11,23 +11,46 @@ integrate it into your workflow.
+A default configuration is included in the package which contains options for loading different file formats. This is a
+[YAML](https://yaml.org) file with keys for each file type and the relevant key/value pair nested within. The table
+below details the current configuration options. If loading a file format supports a custom configuration file then it
+can be specified with the `config_path` parameter which should point to the location of the file.
```
How is `config_path` used to specify an alternative config file?

ns-rse replied:
Its a parameter to `load_jpk()` and so you would...

```
from AFMReader import jpk


some_jpk_file = jpk.load_jpk(file_path="path/to/some.jpk", channel="height_trace", config_path="path/to/config.yaml")
```

AFMReader, whilst used by TopoStats, is its own stand alone package and can be used in isolation. By introducing the option here it is making it available for users and in the long run TopoStats might leverage but I've not got as far as integrating that because I've other priorities.  If and when the need arises we can look at doing that, but setting things up in this flexible manner now makes that easier if its needed in the future.

derollins replied:
Thanks, I re-read the `_load_jpk_tags()` and understand now. Makes sense.
